pkg/gitrepository/manager.go

```diff
@@ -0,0 +1,139 @@
+package gitrepository
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//Manager is responsible for managing gitrepository resources
+type Manager struct {
+	kClient   Kubernetes
+	namespace string
+	ctx       context.Context
+	timeout   time.Duration
+	interval  time.Duration
+}
+
+//Instance contains a tag and path of profile.yaml
+type Instance struct {
+	Tag  string
+	Path string
+}
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+//counterfeiter:generate -o fakes/fake_kubernetes.go . Kubernetes
+//Kubernetes interface for itneracting with kubernetes
+type Kubernetes interface {
+	Get(ctx context.Context, key client.ObjectKey, obj client.Object) error
+	Create(ctx context.Context, obj client.Object, opts ...client.CreateOption) error
+	Delete(ctx context.Context, obj client.Object, opts ...client.DeleteOption) error
+}
+
+//NewManager returns a Manager struct
+func NewManager(ctx context.Context, namespace string, kClient Kubernetes, timeout, interval time.Duration) *Manager {
+	return &Manager{
+		kClient:   kClient,
+		namespace: namespace,
+		ctx:       context.TODO(),
```

Comment:
you aren't using the passed in context. This should be `ctx`.

Reply:
oops good spot!